persistence-modules/spring-data-jpa-query-2/src/test/java/com/baeldung/persistence/hibernate/FooSortingPersistenceIntegrationTest.java

```diff
@@ -152,9 +171,15 @@ public final void whenCriteriaSortingStringNullsLastAsc_thenNullsLast() {
 
     @Test
     public final void whenCriteriaSortingStringNullsFirstDesc_thenNullsFirst() {
-        final Criteria criteria = session.createCriteria(Foo.class, "FOO");
-        criteria.addOrder(Order.desc("name").nulls(NullPrecedence.FIRST));
-        final List<Foo> fooList = criteria.list();
+        CriteriaQuery<Foo> selectQuery = session.getCriteriaBuilder().createQuery(Foo.class);
+        selectQuery.from(Foo.class);
+        Query<Foo> query = session.createQuery(selectQuery);
+
```

Comment:
Is adding initial capacity (2) needed?

Reply:
if we know beforehand that the initial capacity of the list is 2, it's better to specify it.

Reply:
but it's not needed it will work anyway.